src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1628,42 +1628,60 @@
     (fs/delete csr-path)
     (log/debug (i18n/trs "Removed certificate request for {0} at ''{1}''" subject csr-path))))
 
+(schema/defn prevent-duplicate :- [schema/Int]
```

Comment:
I would call this something a little more specific, maybe something like `filter-already-revoked-serials`?